src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -337,7 +337,7 @@ public function sendHeaders()
         }
 
         // headers
-        foreach ($this->headers->allPreserveCaseWithoutCookies() as $name => $values) {
+        foreach ($this->headers->allPreserveCase() as $name => $values) {
```

Comment:
The current cookie code distinguishes between urlencoding the cookie value and not doing that (raw), the ```setcookie()``` function also performs some filtering and escaping as I understand it. As far as I can see in the code, the ```allPreserveCase()``` function doesn't do that but just casts the ```Cookie``` object to ```string``` and as such isn't a 1-to-1 substitute for ```setcookie()``` and ```setrawcookie()```, right?